hibernate-core/src/main/java/org/hibernate/cache/internal/EnabledCaching.java

```diff
@@ -491,32 +506,26 @@ public QueryResultsCache getQueryResultsCacheStrictly(String regionName) {
 			return null;
 		}
 
+		// namedQueryResultsCacheMap does not include defaultQueryResultsCache
 		return namedQueryResultsCacheMap.get( regionName );
 	}
 
 	protected QueryResultsCache makeQueryResultsRegionAccess(String regionName) {
-		final Region region = regionsByName.computeIfAbsent(
+		final QueryResultsCache regionAccess = namedQueryResultsCacheMap.computeIfAbsent(
 				regionName,
-				this::makeQueryResultsRegion
+				this::makeQueryResultsCache
 		);
-		final QueryResultsRegion queryResultsRegion;
-		if ( QueryResultsRegion.class.isInstance( region ) ) {
-			queryResultsRegion = (QueryResultsRegion) region;
-		}
-		else {
-			// There was already a different type of Region with the same name.
-			queryResultsRegion = queryResultsRegionsByDuplicateName.computeIfAbsent(
-					regionName,
-					this::makeQueryResultsRegion
-			);
-		}
-		final QueryResultsCacheImpl regionAccess = new QueryResultsCacheImpl(
+		domainDataAndQueryResultsRegionNames.add( regionName );
+		legacySecondLevelCacheNames.add( regionName );
```

Comment:
The new named ```QueryResultsRegion``` is not qualified when added to #legacySecondLevelCacheNames. 

Should it be qualified as follows?

```
legacySecondLevelCacheNames.add( StringHelper.qualifyConditionally( 
        getSessionFactory().getSessionFactoryOptions().getCacheRegionPrefix(), 
        regionName ) 
);
```

Reply:
Actually, I believe that the legacy behavior was to not expect the user to prefix the names for query regions which is different than its expectations for domain data regions.  Not completely sure, so we should maybe verify that.